### PR TITLE
Collection literals working group notes 2023-05-10

### PR DESCRIPTION
**Pull request by jnm2**

### CODE REVIEW

**Review by CyrusNajmabadi — APPROVED** [review]

File: meetings/working-groups/collection-literals/CL-2023-05-10.md [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+### Option 2: optimize away `List<T>`
+
+Under this option, the natural type of collection literals would be `List<T>` as previously proposed. A `List<T>` instance would not actually be created if a different backing storage performs better and the difference is not observable by the code.
+
+These performance benefits would extend to all existing code that creates a new `List<T>` instance using syntax. No `List<T>` would be created in code such as the following; instead, this would be lowered to `stackalloc` or similar:
+
+```cs
+List<int> list = new List<int> { 1, 2 };
+foreach (var item in list) // Only usage of 'list'
+{
```
Comment (RikkiGibson) [inline meetings/working-groups/collection-literals/CL-2023-05-10.md:40]:
```suggestion
{
}
```

**Comment by CyrusNajmabadi** [comment]
Nit: you could consider linking to the speclets we went over.

File: meetings/working-groups/collection-literals/CL-2023-05-10.md (hunk 1 above)
Comment (jnm2) [reply meetings/working-groups/collection-literals/CL-2023-05-10.md:40]: Would `// ...` on the next line work as well, without implying that the compiler could have disappeared the entire loop as an optimization? I guess an empty loop body isn't that distracting.

**Comment by jnm2** [comment]
> Nit: you could consider linking to the speclets we went over.

Would you like to come up with a longterm structure in your repo first, now that we're accruing a few files and have separate features ahead of us?

Are there guidelines about cross-repo links? I remember dotnet/runtime having an opinion a while back. Should we copy the speclets into a subfolder within this repo?

File: meetings/working-groups/collection-literals/CL-2023-05-10.md (hunk 1 above)
Comment (RikkiGibson) [reply meetings/working-groups/collection-literals/CL-2023-05-10.md:40]: Main reason I suggested is, the highlighting is getting totally confused after this, and I figured this was the reason :)

Comment (jnm2) [reply meetings/working-groups/collection-literals/CL-2023-05-10.md:40]: Oh, great point :D

**Pull request merged**